barretenberg/cpp/src/barretenberg/vm2/simulation/tx_execution.hpp

```diff
@@ -18,8 +18,15 @@ class TxExecution final {
 
     void simulate(const Tx& tx);
 
+    std::unique_ptr<ContextInterface> make_enqueued_context(AztecAddress address,
+                                                            AztecAddress msg_sender,
+                                                            std::span<const FF> calldata,
+                                                            bool is_static);
+
   private:
     ExecutionInterface& call_execution;
+    // More things need to be lifted into the tx execution??
```

Comment:
Won't you need to lift the whole context? (that would include the db)

Reply:
Possibly, I'm not sure how the tx-level operations might utilise the context at the moment, outside of just mutating the initial one it creates.